Packages/com.deltation.toon-rp/Runtime/PostProcessing/BuiltIn/ToonScreenSpaceOutlineComponent.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using UnityEngine;
+using UnityEngine.Rendering;
+using static DELTation.ToonRP.PostProcessing.BuiltIn.ToonScreenSpaceOutlineSettings;
+
+namespace DELTation.ToonRP.PostProcessing.BuiltIn
+{
+    [VolumeComponentMenu("ToonRP/Toon Screen Space Outlines")]
+    public class ToonScreenSpaceOutlineComponent : VolumeComponent
+    {
+        [Header("General")]
+        public ColorParameter OutlineColor = new ColorParameter(Color.black);
+        public BoolParameter UseFog = new BoolParameter(true);
+        public MinFloatParameter MaxDistance = new MinFloatParameter(100f, 0f);
+        public ClampedFloatParameter DistanceFade = new ClampedFloatParameter(0.1f, 0.001f, 1f);
+
+
+        [Header("Filters")]
+        public OutlineFilterParameter ColorFilter = new OutlineFilterParameter(new OutlineFilterReduced());
+        public OutlineFilterParameter DepthFilter = new OutlineFilterParameter(new OutlineFilterReduced());
+        public OutlineFilterParameter NormalsFilter = new OutlineFilterParameter(new OutlineFilterReduced());
+        
+        // public BoolParameter boolParameter = new BoolParameter(false);
+
+        public bool IsActive() => OutlineColor.value.a > 0f;
+
+        protected override void OnEnable()
+        {
+            base.OnEnable();
+            displayName = "Toon Screen Space Outlines";
+        }
+    }
+
+    /// <summary>
+    /// Simple container for outline filter settings
+    /// </summary>
+    [System.Serializable]
+    public class OutlineFilterParameter : VolumeParameter<OutlineFilterReduced>
+    {
+        public OutlineFilterParameter(OutlineFilterReduced value, bool overrideState = false)
+            : base(value, overrideState) { }
+
+        /*
+        public override void Interp(OutlineFilterReduced from, OutlineFilterReduced to, float t)
+        {
+            m_Value.Enabled = from.Enabled;
+            m_Value.Threshold = Mathf.Lerp(from.Threshold, to.Threshold, t);
+            m_Value.Smoothness = Mathf.Lerp(from.Smoothness, to.Smoothness, t);
+            
+        }*/
+    }
+    [Serializable]
+    public class OutlineFilterReduced
+    {
+        /// <summary>
+        /// Used only for passing data from default settings
+        /// </summary>
+        /// Since this is true it will not have any effect on the filter asset, it also allows easier casts
+        [HideInInspector]
+        public bool FilterEnabled = true;
```

Comment:
It if does not affect anything, and the casts are done through field-by-field copying, do we really need it here?